products/views.py

```diff
@@ -1,3 +1,28 @@
-from django.shortcuts import render
+from django.http  import JsonResponse
+from django.views import View
+
+from products.models import Product, ProductImage, Option
+
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        try:
+            product = Product.objects.get(id=product_id)
+            images  = ProductImage.objects.filter(product_id=product.id)
+            options = Option.objects.all()
+
+            data = {
+                    'image'       : [product_image.image_url for product_image in images],
+                    'name'        : product.name,
+                    'price'       : product.price,
+                    'description' : product.description,
+                    'option'      : {
+                        'name'  : [product_option.name for product_option in options],
+                        'price' : [product_option.price for product_option in options]
+                    }
+                }
+
+            return JsonResponse({'data':[data]}, status=201)
```

Comment:
```suggestion
            return JsonResponse({'data':data}, status=200)
```
- 굳이 데이터를 다시 list로 감쌀 필요는 없습니다.
- 생성하는 것이 아니기 때문에 status 코드는 200으로 수정해 주세요.

Reply:
넵 알겠습니다!